modules/mauth-authenticator/src/main/scala/com/mdsol/mauth/scaladsl/RequestAuthenticator.scala

```diff
@@ -39,29 +48,73 @@ class RequestAuthenticator(publicKeyProvider: ClientPublicKeyProvider, epochTime
+              case MAuthVersion.MWS =>
+                validateSignatureV1(mAuthRequest, clientPublicKey)
+              case MAuthVersion.MWSV2 =>
+                validateSignatureV2(mAuthRequest, clientPublicKey)
+            }
         }
       )
     }
     promise.future
   }
 
+  /**
+    * check if mauth v1 is disabled or not
+    * @return True or false identifying if mauth v1 is disabled or not.
+    */
+  override def isDisableV1() = disableV1
+
   // Check epoch time is not older than specified interval.
   protected def validateTime(requestTime: Long)(requestValidationTimeout: Duration): Boolean =
     (epochTimeProvider.inSeconds - requestTime) < requestValidationTimeout.toSeconds
+
+  // Check V2 header if V1 is disabled
+  protected def validateMauthVersion(mAuthRequest: MAuthRequest, disableV1: Boolean): Boolean =
+    !disableV1 || mAuthRequest.getMauthVersion == MAuthVersion.MWSV2
+
+  // check signature for V1
+  private def validateSignatureV1(mAuthRequest: MAuthRequest, clientPublicKey: PublicKey): Boolean = {
+    val decryptedSignature = MAuthSignatureHelper.decryptSignature(clientPublicKey, mAuthRequest.getRequestSignature)
+    // Recreate the plain text signature, based on the incoming request parameters, and hash it.
+    val unencryptedRequestString = MAuthSignatureHelper.generateUnencryptedSignature(
+      mAuthRequest.getAppUUID,
+      mAuthRequest.getHttpMethod,
+      mAuthRequest.getResourcePath,
+      new String(mAuthRequest.getMessagePayload, StandardCharsets.UTF_8),
+      String.valueOf(mAuthRequest.getRequestTime)
+    )
+    val messageDigest_bytes = MAuthSignatureHelper.getHexEncodedDigestedString(unencryptedRequestString).getBytes(StandardCharsets.UTF_8)
+
+    // Compare the decrypted signature and the recreated signature hashes.
+    // If both match, the request was signed by the requesting application and is valid.
+    util.Arrays.equals(messageDigest_bytes, decryptedSignature)
+  }
+
+  // check signature for V2
+  private def validateSignatureV2(mAuthRequest: MAuthRequest, clientPublicKey: PublicKey): Boolean = {
+    // Recreate the plain text signature, based on the incoming request parameters, and hash it.
+    val unencryptedRequestString = MAuthSignatureHelper.generateStringToSign(
+      mAuthRequest.getAppUUID,
+      mAuthRequest.getHttpMethod,
+      mAuthRequest.getResourcePath,
+      mAuthRequest.getQueryParameters,
+      new String(mAuthRequest.getMessagePayload, StandardCharsets.UTF_8),
+      String.valueOf(mAuthRequest.getRequestTime),
+      MAuthVersion.MWSV2
+    )
+
+    // Compare the decrypted signature and the recreated signature hashes.
+    try {
+      MAuthSignatureHelper.verifyRSA(unencryptedRequestString, mAuthRequest.getRequestSignature, clientPublicKey)
+    } catch {
+      case ex: Exception =>
+        val message = "MAuth request validation failed because of " + ex.getMessage
+        logger.error(message)
+        throw new MAuthValidationException(message)
```

Comment:
Instead of throwing an exception this method could return an Either, or just logging an returning false?

Reply:
changes as suggested( just logging and returning false)